simulator/include/core.h

```diff
@@ -0,0 +1,42 @@
+#ifndef CHS_CORE_H
+#define CHS_CORE_H
+
+#include <cinttypes>
+#include <vector>
+
+#include "cache.h"
+
+namespace CacheHierarchySimulator
+{
+
+/**
+ * Class for representing a single core.
+ * This class includes a list of cache objects, as some cores have their own caches.
+*/
+class Core
+{
+public:
+
+    struct Stats : Cache::Stats
+    {
+        std::vector<Cache::Stats> cacheStats;  
+    };
+
+    void addCache(const Cache& cache);
+
+    CacheReturn read(Address address);
+    CacheReturn write(Address address);
+
+    Core::Stats getStats();
+
+private:
+
+    Core::Stats stats;
```

Comment:
Unless I am misunderstanding, isn't this variable obsolete? each cache in the caches vector should have its own stats so there shouldn't be a need to keep a separate vector of cache stats right?

Reply:
I see what you mean. My intent was to have a structure to hold the overall statistics of the Core and System classes, and then have them returned as a struct. I see that having the separate vector stored is redundant. I did not think of that so I will fix that.